src/target_intacct/payment_record.py

```diff
@@ -0,0 +1,173 @@
+import pandas as pd
+import datetime
+import singer
+
+from .utils import get_input
+
+from .const import PAYMENT_RECORDS_REQUIRED_COLS, PAYMENT_RECORDS_REQUIRED_CONFIG_KEYS
+
+logger = singer.get_logger()
+
+
+def build_line_items(gross_amount, total_fees, total_sales_tax, config):
+    """Builds the line items for Intacct uploads"""
+    accountno = config["accountno_1"]
+    details = {"memo": config["memo"], "locationid": config["locationid"], "departmentid": config["departmentid"]}
+    lines = [{"glaccountno": accountno, "amount": gross_amount/100, **details}]
+    if total_fees > 0:
+        accountno = config["accountno_2"]
+        lines.append({"glaccountno": accountno, "amount": -(total_fees/100), **details})
+    if total_sales_tax > 0:
+        accountno = config["accountno_3"]
+        lines.append({"glaccountno": accountno, "amount": total_sales_tax/100, **details})
+
+    return lines
+
+def get_date_lines(year, month, day):
+    """Get a formatted date line"""
+    return { "year": year, "month": month, "day": day }
+
+def get_receipt_data(transactions):
+        """Get the total amount, fees, and taxes for a payout's transactions"""
+        gross_amount, total_fees, total_adjustments, total_refunds, total_sales_tax = 0, 0, 0, 0, 0
+
+        mapped_transactions = transactions.groupby("transaction_type")
+        
+        # Handles data by group (so all charges at once, all fees at once, etc)
+        for group_name, group_df in mapped_transactions:        
+            if group_name == "charge":
+                gross_amount += group_df["amount"].sum()
+                total_fees += group_df["fee"].sum()               
+                total_sales_tax += group_df["tax"].sum() 
+
+            elif group_name == "fee":
+                total_fees += group_df["amount"].sum() 
+                total_fees += group_df["fee"].sum()
+            elif group_name == "adjustment":
+                total_adjustments += group_df["amount"].sum()
+                total_fees += group_df["fee"].sum()
+            elif group_name == "refund":
+                total_refunds += group_df["amount"].sum()
+                total_fees += group_df["fee"].sum()
+
```

Comment:
is there a scenario where the group_name does not match 'charge'/'fee'/'adjustment'/'refund' ?
If those capture all of them, then the line:
`total_fees += group_df["fee"].sum()` 
could be pulled to before all the ifs.

Reply:
hmmm unfortunately that is a possibility (not in our data pipeline but it could be in others). I've added some logging to handle any other group_names that might come up